### PR TITLE
Chore: Implement barrel pattern

### PR DESCRIPTION
**Pull request by saszaCodes**
#### What does this PR do?
Implements barrel pattern in our project
#### How should this be manually tested?
There is no other method than to look at the files and check if they are structured correctly and if imports / exports actually import / export what they should or not

### CODE REVIEW

File: src/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import logger from './logger';
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (Avuidrauxs) [inline src/lib/index.js]: ### Nitpick

Disable this in the eslint file rather.

File: src/app.js [hunk 2]
```diff
@@ -1,23 +1,25 @@
 import express from 'express';
-import morganMiddleware from './loggingMiddleware/morgan';
-import errorHandler from './errorHandler';
+import { errorHandlers, logging } from './middleware';
+// import middleware from './middleware';
 
+// const { errorHandlers } = middleware;
```
Comment (Avuidrauxs) [inline src/app.js]: ### Nitpick

Remove unused commented code

File: src/middleware/index.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import errorHandlers from './errorHandlers';
+import logging from './logging';
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (Avuidrauxs) [inline src/middleware/index.js]: ### Nitpick

Lets disable it in the eslint config

File: src/middleware/logging/index.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import requestLogger from './requestLogger';
+
+const logging = {
+  requestLogger
+};
+
+// eslint-disable-next-line import/prefer-default-export
```
Comment (Avuidrauxs) [inline src/middleware/logging/index.js]: ### Nitpick

Lets disable it in the eslint config

File: src/middleware/errorHandlers/index.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import generic from './generic';
+
+const errorHandlers = {
+  generic,
+};
+
+export default errorHandlers;
```
Comment (Avuidrauxs) [inline src/middleware/errorHandlers/index.js:7]: ### Comment

This is a decent way of exporting also.

**Review by Avuidrauxs — CHANGES_REQUESTED** [review]
Good job, just some small nitpicks and we're good to go.

**Comment by saszaCodes** [comment]
@Avuidrauxs ok, I implemented suggested changes

**Review by Avuidrauxs — APPROVED** [review]
LGTM!

**Pull request merged**